src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DrawSpecialPanelAdapter.java

```diff
@@ -0,0 +1,41 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+/**
+ * driver adapter to drawer with several bugs.
+ */
+public class DrawSpecialPanelAdapter extends DrawPanelController implements Job2dDriver {
+    private int startX = 0, startY = 0;
+
+    public DrawSpecialPanelAdapter() {
+        super();
+    }
+
+    @Override
+    public void setPosition(int x, int y) {
+        this.startX = x;
+        this.startY = y;
+    }
+
+    @Override
+    public void operateTo(int x, int y) {
+        ILine line;
+        line = LineFactory.getSpecialLine();
```

Comment:
Unfortunately, this is not a generic solution, for each type of line another class is required.